app/src/main/java/com/example/sharingang/SetPriceActivity.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.sharingang
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.EditText
+
+class SetPriceActivity : AppCompatActivity() {
+    companion object {
+        const val EXTRA_PRICE = "com.example.sharingang.EXTRA_PRICE"
+        var CHOSEN_CURRENCY = "USD"
+    }
+
+    private val defaultPrice = 0.0
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_set_price)
+    }
+
+    fun setPriceSummary(view: View) {
+        val intent = Intent(this, SetPriceSummaryActivity::class.java)
+        val editTextPrice : EditText = findViewById(R.id.editTextSetPrice)
+        val price =
+            if(editTextPrice.text.isNullOrBlank()) defaultPrice // if no price has been entered
+            else editTextPrice.text.toString().toDouble()
+
+        intent.putExtra(EXTRA_PRICE, "The price you set is: $price $CHOSEN_CURRENCY.")
```

Comment:
I don't think it's great that you're choosing how to display the information by sending a formatted string through the intent.

I think it would make more sense to send out a double representing the price instead. Perhaps combine one extra for the amount, and another for the currency, if you want to support multiple currencies. Regardless, sending out a pre-formatted string seems like a bad idea, because the activity displaying the price should be in control of how it does that.

Reply:
Okay! So instead of :
`intent.putExtra(EXTRA_PRICE, "The price you set is: $price $CHOSEN_CURRENCY.")`
I can have something like
`intent.putExtra(EXTRA_PRICE, price)`?
or even
`intent.putExtra(EXTRA_PRICE, Pair(price, CHOSEN_CURRENCY))` ?

Reply:
Yeah, that's what I'm getting at!